src/crypto/algorithms/megolm.js

```diff
@@ -543,6 +547,9 @@ MegolmDecryption.prototype.decryptEvent = function(event) {
     } catch (e) {
         if (e.message === 'OLM.UNKNOWN_MESSAGE_INDEX') {
             this._addEventToPendingList(event);
+            if (requestKeysOnFail) {
+                this._requestKeysForEvent(event);
+            }
```

Comment:
We don't recover at this point, we just throw as if `requestKeysOnFail == false` so from the caller's perspective they have no idea whether they will ever be able to decrypt this message. This might be fine? I don't know how people are using `MegolmDecryption.prototype.decryptEvent` currently.